src/array.cr

```diff
@@ -2022,14 +2038,26 @@ class Array(T)
       l = a + i
       v = l.value
       p = l - 1
-      while l > a && comp.call(v, p.value) < 0
+      while l > a && cmp(v, p.value, comp) < 0
         l.value = p.value
         l, p = p, p - 1
       end
       l.value = v
     end
   end
 
+  protected def self.cmp(v1, v2)
+    v = v1 <=> v2
+    raise ArgumentError.new("comparison of #{v1} and #{v2} failed") if v.nil?
```

Comment:
In exception messages title-case if preferred. Ditto below.

Reply:
I wish we hand't change that. This is how it looks right now:

```
Unhandled exception: Index out of bounds (IndexError)
  from src/indexable.cr:596:8 in 'at'
  from src/indexable.cr:73:5 in '[]'
  from baz.cr:2:1 in '__crystal_main'
  from src/crystal/main.cr:97:5 in 'main_user_code'
  from src/crystal/main.cr:86:7 in 'main'
  from src/crystal/main.cr:106:3 in 'main'
```

It doesn't look good to have an uppercase letter after `:`. This is Ruby:

```
baz.cr:1:in `sort': comparison of Integer with 2 failed (ArgumentError)
	from baz.cr:1:in `<main>'
```

I'll capitalize it, but I'd like to discuss reverting this capitalization.